lib/widgets/recent_dm_conversations.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+
+import '../model/narrow.dart';
+import '../model/recent_dm_conversations.dart';
+import 'content.dart';
+import 'icons.dart';
+import 'message_list.dart';
+import 'page.dart';
+import 'store.dart';
+import 'text.dart';
+
+class RecentDmConversationsPage extends StatefulWidget {
+  const RecentDmConversationsPage({super.key});
+
+  static Route<void> buildRoute({required BuildContext context}) {
+    return MaterialAccountPageRoute(context: context,
+      builder: (context) => const RecentDmConversationsPage());
+  }
+
+  @override
+  State<RecentDmConversationsPage> createState() => _RecentDmConversationsPageState();
+}
+
+class _RecentDmConversationsPageState extends State<RecentDmConversationsPage> with PerAccountStoreAwareStateMixin<RecentDmConversationsPage> {
+  RecentDmConversationsView? model;
+
+  @override
+  void onNewStore() {
+    model?.removeListener(_modelChanged);
+    model = PerAccountStoreWidget.of(context).recentDmConversationsView
+      ..addListener(_modelChanged);
+  }
+
+  void _modelChanged() {
+    setState(() {
+      // The actual state lives in [model].
+      // This method was called because that just changed.
+    });
+  }
+
+  Widget _buildItem(BuildContext context, DmNarrow narrow) {
+    final allRecipientIds = narrow.allRecipientIds;
+    final store = PerAccountStoreWidget.of(context);
+    final selfUser = store.users[store.account.userId]!;
+    final recipientsSansSelf = allRecipientIds
+      .whereNot((id) => id == selfUser.userId)
+      .map((id) => store.users[id]!)
```

Comment:
This `!` — so that we crash if a user appears which we don't know about — reflects how we'd write this sort of logic in zulip-mobile, but I'd prefer to soften these failures while we're rewriting things here.

In particular since we're planning a product feature where guests (for example) might not know about all the users in the realm, it'll start becoming a lot more likely that the server could have a bug, or even perhaps an edge case that's hard to eliminate even when we fully understand it, where we end up with a user in a list like this that isn't a user we know about. We'll want to try to avoid those situations, but it'd be good to avoid crashing in them.

See [MessageListAppBarTitle] for an example of handling that. Basically we'll end up with things like `?? '(unknown user)'` sprinkled around.

Reply:
Ah, okay.